sdccc/src/test/java/com/draeger/medical/sdccc/tests/util/MdibHistorianTest.java

```diff
@@ -855,54 +865,206 @@ void testUniqueEpisodicReportBasedHistoryGoodDoNotFilterWhenReportTypeDiffers()
+    void testApplyReportOnStorageRegressionCalledWithOperationInvokedReport()
+            throws ReportProcessingException, PreprocessingException {
+
+        // given
+        final BigInteger numericMdibVersion = BigInteger.ZERO;
+        final String sequenceId = "abc";
+        final var report = new OperationInvokedReport();
+        final var reportParts = new ArrayList<OperationInvokedReport.ReportPart>();
+        reportParts.add(new OperationInvokedReport.ReportPart());
+        report.setReportPart(reportParts);
+        report.setMdibVersion(numericMdibVersion);
+        report.setSequenceId(sequenceId);
+
+        testApplyReportOnStorage(numericMdibVersion, sequenceId, report);
+    }
 
-        final var contextReportMdibVersion = BigInteger.ONE;
-        final var contextReportStateVersion = BigInteger.TWO;
-        final var contextReport = buildEpisodicContextReport(
-                MdibBuilder.DEFAULT_SEQUENCE_ID, contextReportMdibVersion, contextReportStateVersion);
+    /**
+     * Tests if applyReportOnStorage() can gracefully ignore OperationInvokedReports.
+     * NOTE: this is a regression test: before it was fixed, applyReportOnStorage() did fail with an Exception
+     *       in this case.
+     * @throws ReportProcessingException - when applyReportOnStorage() throws it
+     * @throws PreprocessingException - when applyReportOnStorage() throws it
+     */
+    @Test
+    void testApplyReportOnStorageRegressionCalledWithSystemErrorReport()
+            throws ReportProcessingException, PreprocessingException {
+
+        // given
+        final BigInteger numericMdibVersion = BigInteger.ZERO;
+        final String sequenceId = "abc";
+        final var report = new SystemErrorReport();
+        final SystemErrorReport.ReportPart part = new SystemErrorReport.ReportPart();
+        part.setErrorCode(createCodedValue("someCode", "someCodingSystem", "1.0"));
+        final LocalizedText text =
+                createLocalizedText("someErrorText", "en", "someRef", BigInteger.ZERO, LocalizedTextWidth.M);
+        part.setErrorInfo(text);
+        report.getReportPart().add(part);
+        report.setMdibVersion(numericMdibVersion);
+        report.setSequenceId(sequenceId);
+
+        testApplyReportOnStorage(numericMdibVersion, sequenceId, report);
+    }
 
-        final var firstMod =
-                (EpisodicContextReport) contextReport.getBody().getAny().get(0);
-        firstMod.getReportPart().get(0).getContextState().add(operator1);
-        firstMod.getReportPart().get(0).getContextState().add(operator2);
+    @Test
+    void testUniqueEpisodicReportBasedHistoryUntilTimestampRegressionWithOperationInvokedReport()
+            throws ReportProcessingException, JAXBException, PreprocessingException, IOException {
+
+        final BigInteger numericMdibVersion = BigInteger.ZERO;
+        final String sequenceId = "abc";
+
+        final var messageObjectFactory = new ObjectFactory();
```

Comment:
You do not need to instantiate the factory here, you can simply use the MessageBuilder.